torchbenchmark/models/opacus_cifar10/__init__.py

```diff
@@ -0,0 +1,81 @@
+import torch
+import torch.optim as optim
+import torch.nn as nn
+import torchvision.models as models
+from opacus.utils.module_modification import convert_batchnorm_modules
+from opacus import PrivacyEngine
+
+from ...util.model import BenchmarkModel
+
+
+class Model(BenchmarkModel):
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+
+        self.model = models.resnet18(num_classes=10)
+        self.model = convert_batchnorm_modules(self.model)
+        self.model = self.model.to(device)
+
+        self.example_inputs = (
+            torch.randn((64, 3, 32, 32), device=self.device),
+        )
+        self.example_target = torch.randint(0, 10, (64,), device=self.device)
+
+    def get_module(self):
+        if self.jit:
+            raise NotImplementedError()
+
+        return self.model, self.example_inputs
+
+    def train(self, niter=3):
+        if self.jit:
+            raise NotImplementedError()
+
+        model, (images,) = self.get_module()
+        targets = self.example_target
+
+        optimizer = optim.Adam(model.parameters(), lr=0.001)
+        criterion = nn.CrossEntropyLoss()
+
+        # This is supposed to equal the number of data point.
+        # It is only to compute stats so dwai about the value.
+        sample_size = 64 * 100
+        clipping = {"clip_per_layer": False, "enable_stat": False}
+
+        privacy_engine = PrivacyEngine(
```

Comment:
is constructing the PrivacyEngine something you need to do every step?  I would suggest moving this part to __init__ and focusing 'train' on just the `for _  in range` part.

Reply:
Hmm, privacy engine should just be constructed once. It needs to accept `optim` as an argument. Is it fine for the optimizer construction to also be in `__init__`?

Reply:
yea, the more the better (moved to __init__).  the stuff that should be in train() is typically just calling forward, backward, step.